rs/src/write.rs

```diff
@@ -1,18 +1,20 @@
 use std::fmt::Display;
 
 use crate::core::Dependency;
-use crate::graph::Graph;
 
 pub enum TextFormat {
     Simple,
     Full,
 }
 
-pub struct Text<'a, V>
+pub struct Text<'g, G, V, It, InnIt>
 where
-    V: Clone + Ord,
+    G: IntoIterator<IntoIter = It>,
+    V: Clone + PartialEq + 'g,
+    It: Iterator<Item = (&'g V, InnIt)>,
+    InnIt: Iterator<Item = (&'g V, Dependency)>,
```

Comment:
Suspect you might be able to simplify these bounds:

```suggestion
    G: IntoIterator<Item = (&'g V, InnIt)>,
    V: Clone + PartialEq + 'g,
    InnIt: Iterator<Item = (&'g V, Dependency)>,
```